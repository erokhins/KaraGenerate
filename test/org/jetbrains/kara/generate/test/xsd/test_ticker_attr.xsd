<?xml version='1.0' encoding='utf-8' ?>
<xsd:schema xmlns='test' xmlns:xsd='http://www.w3.org/2001/XMLSchema' version="1.0" targetNamespace='test'>

    <xsd:element name="complex">
        <xsd:complexType>
            <xsd:attribute name="ticker1">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:NMTOKEN">
                        <xsd:enumeration value="ticker1"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>

            <xsd:attribute name="ticker2">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:string">
                        <xsd:enumeration value="ticker2"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>

            <xsd:attribute name="noTicker1">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:string">
                        <xsd:enumeration value="ticker1"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>

            <xsd:attribute name="noTicker2">
                <xsd:simpleType>
                    <xsd:restriction base="xsd:string">
                        <xsd:enumeration value="noTicker2"/>
                        <xsd:enumeration value="noTicker3"/>
                    </xsd:restriction>
                </xsd:simpleType>
            </xsd:attribute>

        </xsd:complexType>
    </xsd:element>

</xsd:schema>