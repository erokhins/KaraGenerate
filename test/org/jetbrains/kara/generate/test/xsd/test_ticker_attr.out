AttributeTypeDecl:
   AttributeTypeDecl = (noTicker1, null, enumType)
   Values = [ticker1]

   AttributeTypeDecl = (noTicker2, null, enumType)
   Values = [noTicker2, noTicker3]

AttributeDecl:
   AttributeDecl = (noTicker1, null, null})
   AttributeTypeDecl = (noTicker1, null)

   AttributeDecl = (noTicker2, null, null})
   AttributeTypeDecl = (noTicker2, null)

   AttributeDecl = (ticker1, null, null})
   AttributeTypeDecl = (ticker, null)

   AttributeDecl = (ticker2, null, null})
   AttributeTypeDecl = (ticker, null)

AttributeGroupDecl:
ElementDecl:
   ElementDeclaration = (complex, false)
   ElementGroups = []
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =
      AttributeDecl = (noTicker1, null, null})
      AttributeTypeDecl = (noTicker1, null)

      AttributeDecl = (noTicker2, null, null})
      AttributeTypeDecl = (noTicker2, null)

      AttributeDecl = (ticker1, null, null})
      AttributeTypeDecl = (ticker, null)

      AttributeDecl = (ticker2, null, null})
      AttributeTypeDecl = (ticker, null)


ElementGroup:
